cdap-ui/app/directives/timeline/timeline.js

```diff
@@ -16,177 +16,221 @@
-      }
-    },
-  ];
-
-  //Convert all of the dates to the standard date format
-  timelineData.map( (data) => {
-    data.time = new Date(data.time);
-    return data;
-  });
-
-  //Contains only the 'WARN' and 'ERROR' events
-  let filteredEvents = timelineData.filter(function(obj) {
-    if(obj.level === 'WARN' || obj.level === 'ERROR'){
-      return true;
-    }
-    return false;
-  });
+  let timelineData = scope.timelineData;
 
-  let minDate = filteredEvents[0],
-      maxDate = filteredEvents[filteredEvents.length-1];
-
-  //Global Variables
+  //Globals
   let width = element.parent()[0].offsetWidth;
   let height = 50;
+  let paddingLeft = 15;
+  let paddingRight = 15;
+  let maxRange = width - paddingLeft - paddingRight;
+  let sliderLimit = maxRange + 24;
+  let pinX = width-8;
+  let sliderX = 0;
+  let timelineStack = {};
+  let leftVal = 0;
 
-  //Plot function call
-  plot();
-
-  function plot() {
+  //Componenets
+  let leftHandle;
+  let pinHandle;
+  let brush;
+  let xScale;
 
-    //Define SVG
+  scope.plot = function plot() {
+    // -----------------Define SVG and Plot Circles-------------------------- //
     let svg = d3.select('.timeline-log-chart')
                 .append('svg')
                 .attr('width', width)
                 .attr('height', height);
+    //Set the Range and Domain
+    xScale = d3.time.scale().range([0, (maxRange)]);
+    xScale.domain(d3.extent(timelineData, function(d) {
+      return d.time;
+    }));
+    //Define the axes and ticks
+    let xAxis = d3.svg.axis().scale(xScale)
+        .orient('bottom')
+        .innerTickSize(-40)
+        .tickPadding(7)
+        .ticks(8);
```

Comment:
I'm pretty sure this doesn't work. Ticks on time.scale does not accept integer. 

On another note, I believe the design spec wants this to be 12 ticks if I remember my conversation with @bdmogal correctly